compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java

```diff
@@ -65,8 +65,9 @@ public BIRBinaryWriter(BIRNode.BIRPackage birPackage) {
 
     public byte[] serialize() {
         ByteBuf birbuf = Unpooled.buffer();
+        ByteBuf scopebuf = Unpooled.buffer();
```

Comment:
We only need the new buffer at function writing ? why it is needed here?

Reply:
hmmm, looks like we have a new BIRInstructionWriter instance being created, which is not needed. Ideally, an BIRInstructionWriter is needed when writing the function body. I think the use for that is to call the writePosition method in that class. 
Can you just remove this new BIRInstructionWriter creation line here and use the writePosition method found in the BIRBinaryWriter class?

Reply:
Fixed